.travis.yml

```diff
@@ -0,0 +1,38 @@
+language: python
+python:
+  - "2.7"
+before_install:
+  - sudo apt-get update -qq
+  - sudo apt-get install mpi-default-bin mpi-default-dev
+
+  # LAMMPS
+  #  - build lammps as shared library
+  #  - and install LAMMPS python wrapper
+  - git clone git://git.lammps.org/lammps-ro.git mylammps
+  # checkout a recent stable version (from 28 June 2014)
+  - cd mylammps/src && git checkout r12164
+  # make shared library for python
+  - make makeshlib && make -f Makefile.shlib ubuntu_simple
+  - cd ../python && sudo python install.py /usr/lib/ $VIRTUAL_ENV/lib/python2.7/site-packages/
+  # make lammps executable and add to path
+  - cd ../src
+  - make ubuntu_simple
+  - ln -s lmp_ubuntu_simple lammps
+  - export PATH=$TRAVIS_BUILD_DIR/mylammps/src/:$PATH
+  - cd $TRAVIS_BUILD_DIR
+
+  # use current master of simphony
+  - pip install git+git://github.com/simphony/simphony-common.git#egg=simphony
```

Comment:
> If I understand correctly, the splitting of the datacontainers to one for "SD" and others for SP, CM, and BC is mainly due to the serialisation of the SD? 
if this is so, why cannt we just keep one list of CUBA, but implement the serialization of the SP, CM, and BC differently? sorry if I missunderstood something...

It is true that serialization can happen in an number of ways. The main problem there is that if SD, SP, CM and BC all support the same keywords we will need to add an additional check when serializing SD to not consider the unused keywords. The same checks need to take place when generating the Table description but this has to happen anyway.

Another possible problem is memory usage. If the data container does not have a subtype to know which keywords are supported then there is nothing stopping the end user setting unused keys into an SD container e.g.

```python
data = DataContainer()
data[CUBA.PAIR_POTENTIAL] = 'this is a string'
for i in range(100):
    point = Point(data=data)
```
Will end up coping the unused keywords 100 times. While the above is an extreme example the fact that one can by mistake reuse a CM container for SD without the need to filter out unused keywords makes the design less elegant

Another problem is perfomance issues. The DataContainer in a number of places iterates over all the CUBA keywords defined in the CUBA enum this iteration depends on the length of the CUBA enum. As a result there is a small but measurable impact  of having unused keywords in the enum that is used by the DataContainer

Finally I think it would be confusing for a user to see that a SD data container accepts configuration keywords. One would wonder if they are actually necessary. I would prefer a design where we do not have to provide additional documentation that these keywords while allowed in the container are ignored.